web/src/ResourceNav.tsx

```diff
@@ -62,14 +71,34 @@ export function ResourceNavProvider(
     invalidResource = candidateResource
   }
 
+  let search = location.search
+
+  useEffect(() => {
+    let existing = filterByResource[selectedResource] || ""
+    if (existing != search) {
+      let obj = {} as { [key: string]: string }
+      Object.assign(obj, filterByResource)
+      obj[selectedResource] = search
+      setFilterByResource(obj)
+    }
+  }, [selectedResource, search])
+
```

Comment:
(nit) one thing i noticed is that if there's a filter term that's saved on a resource and then the filter state is reset (that is, no filters are applied), this is still saving the empty query string `level=&source=&term=`.